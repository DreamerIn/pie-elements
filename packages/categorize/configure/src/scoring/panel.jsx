import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import { Checkbox } from '@pie-lib/config-ui';
import { Divider } from '../design/buttons';

export class Panel extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    title: PropTypes.string.isRequired,
    enabled: PropTypes.bool.isRequired,
    onToggleEnabled: PropTypes.func.isRequired,
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node
    ]).isRequired
  };
  static defaultProps = {};
  render() {
    const {
      classes,
      className,
      children,
      title,
      enabled,
      onToggleEnabled
    } = this.props;
    return (
      <div className={classNames(classes.panel, className)}>
        <div className={classes.header}>
          <Typography variant="title">{title}</Typography>
          <Checkbox
            label="Enabled"
            checked={enabled}
            onChange={onToggleEnabled}
          />
        </div>
        <Divider />
        {children}
      </div>
    );
  }
}
const styles = () => ({
  panel: {},
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center'
  }
});
export default withStyles(styles)(Panel);
