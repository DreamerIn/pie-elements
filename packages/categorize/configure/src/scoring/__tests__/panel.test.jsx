import { shallow } from 'enzyme';
import React from 'react';

import { Panel } from '../panel';

describe('Panel', () => {
  let w;
  let onToggleEnabled = jest.fn();
  const wrapper = extras => {
    const defaults = {
      classes: {
        panel: 'panel'
      },
      className: 'className',
      title: 'Title',
      enabled: true,
      onToggleEnabled
    };
    const props = { ...defaults, ...extras };
    return shallow(<Panel {...props}>content</Panel>);
  };
  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });
  describe('logic', () => {});
});
