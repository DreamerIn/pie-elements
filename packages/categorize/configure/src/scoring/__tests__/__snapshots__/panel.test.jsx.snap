// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel snapshot renders 1`] = `
<div
  className="panel className"
>
  <div>
    <WithStyles(Typography)
      variant="title"
    >
      Title
    </WithStyles(Typography)>
    <Component
      checked={true}
      label="Enabled"
      onChange={[MockFunction]}
    />
  </div>
  <WithStyles(Component) />
  content
</div>
`;
